### DISCUSSION TITLE
Proper way to handle origin/coordinates mismatch?

### DISCUSSION THREAD

**Discussion by eliasdaler** [jrouwe/JoltPhysics#1020] [Q&A] [answered]
Hello.

When in my engine, the origin of all objects is usually located at the bottom of the model. When I create and position the shape, the center of this shape is mismatched (obviously):
![image](https://github.com/jrouwe/JoltPhysics/assets/1285136/c94cc277-0538-438f-a41d-1e380db86c7d)

Is using `RotatedTranslatedShape` the right way to handle this mismatch?

Basically, I do this:

```cpp
shape = new JPH::CylinderShape(2.6f, 1.8f);
shape = new JPH::RotatedTranslatedShape({0.f, 2.6f, 0.f}, JPH::Quat().sIdentity(), shape);
```

When I do this, the shape works as expected and Jolt's and my engine's positions match 1 to 1.

![image](https://github.com/jrouwe/JoltPhysics/assets/1285136/95ef1c6b-e41a-4436-8fde-604fa33d8704)

So, I was wondering - should I apply these offsets to each mesh/shape which needs it  or is there some other way to handle this?

**Comment by jrouwe** [comment, marked as answer]
> Is using RotatedTranslatedShape the right way to handle this mismatch?

Yes it is

**Reply by eliasdaler** [reply]
I see, thanks!

So, to check that I understand correctly: moving using `RotatedTranslatedShape` just moves the "origin" around which `GetPosition`, `GetCenterOffMassPosition` etc. report will report their values, but it doesn't change anything else about the simulation/shape itself, right?

**Reply by jrouwe** [reply]
Yes, it doesn't change the simulation

**Reply by eliasdaler** [reply]
I see, thanks again!
